### ISSUE TITLE
Tokenized dataset?

### ISSUE THREAD

**Issue by joelburget** [allenai/OLMoE#10]
I was wondering if it'd be possible to upload the tokenized dataset. I tried following the instructions under the Pretraining header but had trouble installing Megablocks due to a CUDA version mismatch. Anyway, I think it would be very helpful to upload the tokenized dataset to Huggingface to save others the work.

**Comment by Muennighoff** [comment]
Agree that this would be great; @soldni what do you think? [Here](https://github.com/allenai/OLMoE/blob/3fc94bf84ebaa3ba33d4d748bfcb67a38c328733/configs/ablations/olmoe-8x1b-newhp-newds-final-s3.yml#L221) are all the s3 paths of the tokenized ds, can we easily upload them to HF?

**Comment by benja-matic** [comment]
Yeah @Muennighoff @soldni , I saw [another issue](https://github.com/allenai/OLMoE/issues/23) mentioning that tokenization takes a long time on a dataset of this size, and I'm experiencing similar issues. If there's any way to upload the tokenized data to hf or to a publicly accessible s3 bucket or to provide presigned URLs to s3 that would be a big help. Appreciate any input you might have.

**Comment by benja-matic** [comment]
> Yeah [@Muennighoff](https://github.com/Muennighoff) [@soldni](https://github.com/soldni) , I saw [another issue](https://github.com/allenai/OLMoE/issues/23) mentioning that tokenization takes a long time on a dataset of this size, and I'm experiencing similar issues. If there's any way to upload the tokenized data to hf or to a publicly accessible s3 bucket or to provide presigned URLs to s3 that would be a big help. Appreciate any input you might have.

A follow up in case others find it useful. I was estimating the time for tokenization by tokenizing just the wiki subset, and it looked like it would take forever. But the throughput increased a lot when tokenizing dclm, presumably because you're IO bound when tokenizing a few files, so adding threads doesn't help much, but it does help a lot when there are many more files than threads. I was able to tokenize the whole dataset in roughly a day when using 120 threads.

**Comment by Muennighoff** [comment]
> Agree that this would be great; [@soldni](https://github.com/soldni) what do you think? [Here](https://github.com/allenai/OLMoE/blob/3fc94bf84ebaa3ba33d4d748bfcb67a38c328733/configs/ablations/olmoe-8x1b-newhp-newds-final-s3.yml#L221) are all the s3 paths of the tokenized ds, can we easily upload them to HF?

@aman-17 do you think you could help with making the tokenized s3 data public? People are still asking for it; thanks a lot!
